### PR TITLE
Add count.reduce()

### PR DESCRIPTION
**Pull request by gnsxun**
ref #3611

### CODE REVIEW

**Comment by youknowone** [comment]
are there related tests in `test_itertools.py`?

**Comment by fanninpm** [comment]
> are there related tests in `test_itertools.py`?

`test_count()` has a pickle-related section.

**Review by youknowone — CHANGES_REQUESTED** [review]
This is not compatible to pickle

File: vm/src/stdlib/itertools.rs [hunk 1]
```diff
@@ -303,6 +312,7 @@ mod decl {
             Ok(*times.read())
         }
 
+
```
Comment (youknowone) [inline vm/src/stdlib/itertools.rs:315]:
```suggestion
```

File: vm/src/stdlib/itertools.rs [hunk 2]
```diff
@@ -193,7 +193,16 @@ mod decl {
     }
 
     #[pyimpl(with(IterNext, Constructor))]
-    impl PyItertoolsCount {}
+    impl PyItertoolsCount {
+
```
Comment (youknowone) [inline vm/src/stdlib/itertools.rs:197]:
```suggestion
```

**Review by youknowone — CHANGES_REQUESTED** [review]

**Review by youknowone — APPROVED** [review]
thank you for contributing!

**Pull request merged**